src/Arriba/Arriba.Server/Controllers/ArribaController.cs

```diff
@@ -0,0 +1,159 @@
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Net;
+using System.Runtime.CompilerServices;
+
+namespace Arriba.Server.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ArribaController : ControllerBase
+    {
+        private readonly IArribaManagementService _arribaManagement;
+        private readonly IArribaServerConfiguration _arribaServerConfiguration;
+
+        public ArribaController(IArribaManagementService arribaManagement, IArribaServerConfiguration arribaServerConfiguration)
+        {
+            _arribaManagement = arribaManagement;
+            _arribaServerConfiguration = arribaServerConfiguration;
+        }
+
+        private IActionResult ExecuteAction(Action action, Func<IActionResult> result)
+        {
+            try
+            {
+                action();
+            }
+            catch (Exception ex)
+            {
+                return ExceptionToActionResult(ex);
+            }
+            return result();
+        }
+
+        [HttpGet]
+        public IActionResult GetTables()
+        {
+            return Ok(_arribaManagement.GetTables());
+        }
+
+        [HttpGet("allBasics")]
+        public IActionResult GetAllBasics()
+        {
+            return Ok(_arribaManagement.GetTablesForUser(this.User));
+        }
+
+        [HttpGet("unloadAll")]
+        public IActionResult GetUnloadAll()
+        {
+            if (!_arribaManagement.UnloadAllTableForUser(this.User))
+                return new ForbidResult();
+
+            return Ok($"All tables unloaded");
+        }
+
+        [HttpGet("table/{tableName}/unload")]
+        public IActionResult GetUnloadTable(string tableName)
+        {
+            if (!_arribaManagement.UnloadTableForUser(tableName, this.User))
+                return new ForbidResult();
+
+            return Ok($"Table {tableName} unloaded");
+        }
+
+        [HttpPost("table")]
+        public IActionResult PostCreateNewTable([Required] CreateTableRequest table)
+        {
+            return ExecuteAction(() => _arribaManagement.CreateTableForUser(table, this.User),
+                          () => CreatedAtAction(nameof(PostCreateNewTable), null));
+        }
+
+        [HttpPost("table/{tableName}/addcolumns")]
+        public IActionResult PostAddColumn(string tableName, [FromBody, Required] IList<ColumnDetails> columnDetails)
+        {
+            return ExecuteAction(() => _arribaManagement.AddColumnsToTableForUser(tableName, columnDetails, this.User),
+                          () => CreatedAtAction(nameof(PostAddColumn), "Columns Added"));
+        }
+
+        private IActionResult ExceptionToActionResult(Exception ex)
+        {
+            if (ex is ArribaAccessForbiddenException)
+                if (_arribaServerConfiguration.EnabledAuthentication)
+                    return Forbid();
```

Comment:
What's the difference between line 92 and 94?

Reply:
When the Authentication is disenabled if we use Forbid we end up getting an Exception because the didn't configure and use authorization. The forbid return 403 and redirects the user to the app login page, to avoid the exception I need to check and return Forbid when it is on and 403 when it is not.